### DISCUSSION TITLE
Issues with Clonezilla and WDS...

### DISCUSSION THREAD

**Discussion by x5etHx** [ipxe/ipxe#618] [Q&A] [answered]
Hi All,

I'm not very familiar with these topics so I'll try to put as much information in as i can...

Setup:
i got a windows server with the following roles installed: AD/ DNS/ DHCP/ WDS

i got the following scripts setup:

iPXE.conf:
```
#!ipxe
chain --autofree Boot\iPXE\boot.ipxe.cfg
```

boot.ipxe.cfg:
```
#!ipxe

# Figure out if client is 64-bit capable
cpuid --ext 29 && set arch x64 || set arch x86
cpuid --ext 29 && set archl amd64 || set archl i386

###################### MAIN MENU ####################################

:start
menu iPXE boot menu
item --gap --             -------------------------------Installation ------------------------------
item --key w wds          WDS
item --key l Clonezilla   Clonezilla
item --gap --             ------------------------- Advanced options -------------------------------
item --key c config       Configure settings
item shell                Drop to iPXE shell
item reboot               Reboot computer
item
item --key x exit         Exit iPXE and continue BIOS boot
choose --timeout 0 --default wds selected || goto cancel
set menu-timeout 0
goto ${selected}
:cancel

[...]

:Clonezilla
kernel \Boot\iPXE\Clonezilla\vmlinuz
initrd \Boot\iPXE\Clonezilla\initrd.img
imgargs initrd=initrd.img boot=live username=user union=overlay components noswap noprompt vga=788 keyboard-layouts=en locales=en_US.UTF-8 fetch=tftp://192.168.101.200/boot/iPXE/Clonezilla/filesystem.squashfs
boot
```

when using imgarg i get an access violation in the TFTP server which is strange, since it downloads vmlinuz and initrd.img fine...

![Wireshark-log](https://user-images.githubusercontent.com/64830382/158611096-2dd0eae4-4a5e-4740-a42d-dc1dcedbe23d.jpg)

Error message on the bootet machine is:

![errorwithargs](https://user-images.githubusercontent.com/64830382/158611031-454d557e-63f9-4c82-ada8-c24c34f9b6df.jpg)

when trying with the following:
```
:Clonezilla
kernel \Boot\iPXE\Clonezilla\vmlinuz initrd=initrd.img boot=live username=user union=overlay components noswap noprompt vga=788 keyboard-layouts=en locales=en_US.UTF-8 fetch=tftp://192.168.101.200/boot/iPXE/Clonezilla/filesystem.squashfs
initrd \Boot\iPXE\Clonezilla\initrd.img
```

i get the resultant error message on the machine:

![error](https://user-images.githubusercontent.com/64830382/158611298-d55a2260-6c77-4661-a084-b79967474a4f.jpg)

i also did test to donwload the filesystem.squashfs via TFTP and that works fine.

DHCP policies have been setup
![dhcp policies](https://user-images.githubusercontent.com/64830382/158609665-2fdef6cf-6cc7-488e-b516-d6e3261965e9.PNG)
reg keys have been added to allow "\" and "/"
![REG-Keys](https://user-images.githubusercontent.com/64830382/158609695-fac2907e-81d3-488c-b15c-58d4892c9ceb.PNG)

im kind of stuck and don't know really how to troubleshoot furhter or what to test ...

**Comment by stappersg** [comment]
* https://ipxe.org/err/3d1260
* https://ipxe.org/err/7f04828e

**Comment by NiKiZe** [comment]
In the first variant you somehow trigger download of initrd=initrd
Make sure it is valid and also that there is no chars in the script that shouldn't be there.

Windows / WDS tftp is not a easy beast to play with. But check the script, and especially the requests in wireshark to find the requested filenames.
Also try to remove parts of the script. Remove one line at a time to try and resolve only the current issue and so on.

**Reply by x5etHx** [reply]
Hi NikiZe,

i did try now to only use the shell to avoid any issues with the script, and typed in the following:

```
kernel \Boot\iPXE\Clonezilla\vmlinuz
module \Boot\iPXE\Clonezilla\initrd.img
Chain vmlinuz initrd=initrd.img
boot
```
resulting in the same strange behaviour that it tries to download "vmlinuz" i also tried different iPXE.efi files e.g. one from a tutorial from the web and one that clonezilla itself provides... I'm really puzzled how this gets triggered...

leaving the args away results in the typical kernel panic VFS issue...

**Comment by mcb30** [comment, marked as answer]
@x5etHx Most likely the problem is caused by using the incorrect path separators (`\` instead of `/`).

iPXE will use the standard `/` path separator to pick a default name for the downloaded image.  For example, if you do:
```
kernel /Boot/iPXE/Clonezilla/vmlinuz
module /Boot/iPXE/Clonezilla/initrd.img
```
then you should find that [`imgstat`](https://ipxe.org/cmd/imgstat) will report two downloaded images named `vmlinuz` and `initrd.img`.

With the incorrect path separators (`\`), iPXE will treat the whole string as a single long filename that still includes the backslashes, e.g. `\Boot\iPXE\Clonezilla\initrd.img`.  The download will work (since the Windows server will happily accept the incorrect path separators), but you'll almost certainly find that [`imgstat`](https://ipxe.org/cmd/imgstat) shows the wrong filenames.  Since these filenames are the ones exposed to the kernel's EFI stub via iPXE's virtual filesystem, the kernel won't be able to find a file called `initrd.img`.

The same behaviour is at play when you try your `chain vmlinuz ...` command: since iPXE thinks that it doesn't already have a downloaded image named just `vmlinuz` then it assumes that you are trying to perform a new download.

You can fix this by using the correct (`/`) path separators (or by using the `--name` option to explicitly name the downloaded images.

**Reply by x5etHx** [reply]
yes this did the trick... awesome you are the man. thanks a lot that made my day
